src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -579,33 +572,11 @@ public void ApplyLogsToDataFile(CancellationToken token, TimeSpan timeToWait)
 
                     var unusedJournals = GetUnusedJournalFiles(jrnls, lastProcessedJournal, lastFlushedTransactionId);
 
-                    var needImmediateFsync =
-                        _pendingSync.IsCompleted &&
-                        (_forceDataSync.Lower() ||
-                         Interlocked.Read(ref _totalWrittenButUnsyncedBytes) > 32 * Constants.Size.Megabyte ||
-                         _timeSinceLastImmediateSync.ElapsedMilliseconds > 3 * 60 * 1000);
```

Comment:
Where do we have this logic now? If those aren't relevant could you explain how it's supposed to work?

Reply:
The sync operation run on [idle](https://github.com/ravendb/ravendb/pull/8148/files#diff-87366e0cab8e4af884a9411d055bbc61R72)
 and when it gets to the [configured threshold](https://github.com/ravendb/ravendb/pull/8148/files#diff-87366e0cab8e4af884a9411d055bbc61R126)

Reply:
But the previous threshold was `32 * Constants.Size.Megabyte` while now we have 1GB by default, right?

We also dropped the concept of forcing the sync after flush if we haven't done it since 3 minutes regardless the threshold and idleness? Please check http://issues.hibernatingrhinos.com/issue/RavenDB-10238

Reply:
Sort of, the idea is that the trigger for a sync is the size of the journals, not anything else.
Flush is because of scratch, sync because of journals.

Reply:
We don't have a test for http://issues.hibernatingrhinos.com/issue/RavenDB-10238 but from some reason we didn't hit 32MB threshold then so we might not hit it now.

CC @aviviadi